sdk/formrecognizer/Azure.AI.FormRecognizer/api/Azure.AI.FormRecognizer.netstandard2.0.cs

```diff
@@ -301,17 +301,23 @@ public partial class AnalyzeResult
         public System.Collections.Generic.IReadOnlyList<Azure.AI.FormRecognizer.DocumentAnalysis.DocumentTable> Tables { get { throw null; } }
     }
     [System.Runtime.InteropServices.StructLayoutAttribute(System.Runtime.InteropServices.LayoutKind.Sequential)]
-    public readonly partial struct BoundingRegion
+    public readonly partial struct BoundingRegion : System.IEquatable<Azure.AI.FormRecognizer.DocumentAnalysis.BoundingRegion>
```

Comment:
Curious why we did this? Is this a customer ask where they want to check equality of the region or is this coming from our design guidelines?

Reply:
It's not explicitly written in our guidelines, but we usually override `IEquatable<T>` for small structs (this one in particular was requested by Krzysztof). I don't think customers will directly check for equality here, but implementing this interface is relevant if these are added to collections such as `Dictionary` or `Set`, which shouldn't accept two copies of the same key.